pkg/controllers/kubeconfig/kubeconfig.go

```diff
@@ -149,18 +154,28 @@ func (r *KubeconfigReconciler) reconcileKubeconfig(ctx context.Context, log logr
 		return ctrl.Result{RequeueAfter: 1 * time.Minute}, nil
 	}
 
-	// Generate kubeconfig
-	kubeconfig, err := generateKubeconfig(kubeconfigOptions{
+	kubeconfigOptions := kubeconfigOptions{
 		token:            tokenSecret.Data["token"],
 		caCert:           tokenSecret.Data["ca.crt"],
 		apiServerEnpoint: r.RestCfg.Host,
 		clusterName:      r.clusterName,
-	})
+	}
 
-	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("error generating kubeconfig: %w", err)
+	if err := validateKubeconfigOptions(kubeconfigOptions); err != nil {
+		if errors.Is(err, errTokenEmpty) || errors.Is(err, errCACertEmpty) {
+			// If the vaildation fails with these errors it means
+			// the token secret has not been populated by the kubernetes control plane yet.
+			// Requeue to wait for the token secret to be populated.
+			log.Info("Token secret has not been populated by the control plane yet.")
+			return ctrl.Result{RequeueAfter: 500 * time.Millisecond}, nil
```

Comment:
We should not even need to requeue because of having a watch on the secret, right?

Reply:
Yeah, I was unsure of the watch, wanted to try E2Es with both, I removed it now, let's see if this behaves the way it should as well.

Reply:
The watch works correctly, so let's go without requeueing :)